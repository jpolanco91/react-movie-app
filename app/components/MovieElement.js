import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'react-proptypes';

const MovieElement = ({ imageSource, title, year, id }) => (
  <div className="column is-one-quarter">
    <div className="card">
      <div className="card-image">
        <figure className="image is-4by3">
          <img src={imageSource} alt="" />
        </figure>
      </div>
      <div className="card-content">
        <div className="media-content">
          <Link to={`/movie_trailer/${id}`}>
            <p className="title is-4">{title}</p>
          </Link>
          <p className="subtitle is-6">{year}</p>
        </div>
      </div>
    </div>
  </div>
);

MovieElement.propTypes = {
  imageSource: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
};

export default MovieElement;
