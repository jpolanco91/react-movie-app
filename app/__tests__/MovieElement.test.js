import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MovieElement from '../components/MovieElement';

test('MovieElement snapshot test', () => {
  const c = create(
    <BrowserRouter>
      <MovieElement
        imageSource="http://www.myapi.com/api/image.png"
        title="My Title"
        year={1976}
        id="1"
        movieKey="1"
      />
    </BrowserRouter>,
  );
  expect(c.toJSON()).toMatchSnapshot();
});
