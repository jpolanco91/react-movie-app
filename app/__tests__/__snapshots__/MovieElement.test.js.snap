// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieElement snapshot test 1`] = `
<div
  className="column is-one-quarter"
>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <figure
        className="image is-4by3"
      >
        <img
          alt=""
          src="http://www.myapi.com/api/image.png"
        />
      </figure>
    </div>
    <div
      className="card-content"
    >
      <div
        className="media-content"
      >
        <a
          href="/movie_trailer/1"
          onClick={[Function]}
        >
          <p
            className="title is-4"
          >
            My Title
          </p>
        </a>
        <p
          className="subtitle is-6"
        >
          1976
        </p>
      </div>
    </div>
  </div>
</div>
`;
